R/abm.R

```diff
@@ -907,3 +1006,84 @@ abm_format_rows <- function(t){
 }
+
+
+#' Create pie chart for Open Access data
+#' 
+#' @param df a data frame at the format produced by abm_oa_data()
+#' @return a pie chart object
+#' @import ggplot2 dplyr plotrix
+#' @importFrom graphics pie
+#' @export
+abm_graph_oadata_pie <- function(df){
+  #unpaywall_cols <- c("#F9BC01", "#8D4EB4", "#20E168", "#CD7F32", "#BBBBBB")
+  #kth_cols <- as.vector(palette_kth(4))
+
+    unpaywall_colors <- data.frame("Gold"="#F9BC01",
+                                   "Hybrid"="#8D4EB4",
+                                   "Green"="#20E168",
+                                   "Bronze"="#CD7F32",
+                                   "Closed"="#BBBBBB") %>%
+        rename("Not OA"="Closed")
+    
+    df_oa_graphdata <- df %>%
+        filter(Publication_Year_ch == "Total") %>%
+        select(gold_count, hybrid_count, green_count, bronze_count, closed_count) %>%
+        rename("Gold" = gold_count,
+               "Hybrid" = hybrid_count,
+               "Green" = green_count,
+               "Bronze" = bronze_count,
+               "Not OA" = closed_count)
+
+        percentages <- df %>%
+        filter(Publication_Year_ch == "Total") %>%
+        mutate(gold_count = 100*gold_count/P_tot,
+               hybrid_count = 100*hybrid_count/P_tot,
+               green_count = 100*green_count/P_tot,
+               bronze_count = 100*bronze_count/P_tot,
+               closed_count = 100*closed_count/P_tot) %>%
+        select(gold_count, hybrid_count, green_count, bronze_count, closed_count) %>%
+        t() %>%
+        format(digits=2) %>%
+        t()
+
+    #Remove empty categories
+    df_oa_graphdata <- df_oa_graphdata[,t(df_oa_graphdata)[,1]!=0]
+    percentages <- percentages[,t(percentages)[,1] %>% as.numeric() != 0]
+    unpaywall_colors <- unpaywall_colors %>%
+        names(.) %in% names(df_oa_graphdata) %>%
+        unpaywall_colors[.] %>%
+        t()
+
+  labls <- paste(names(df_oa_graphdata), "\n", percentages, " %", separator="")
+  pie(t(df_oa_graphdata),  labels = c("","","","","",""), col = unpaywall_colors, cex = 0.8, radius = 0.8)
+  pieangles <- floating.pie(x=t(df_oa_graphdata), col = unpaywall_colors)
+  pie.labels(labels = labls, radius = 1.1, angles = pieangles, cex = 0.8)
+}
+
+
+#' Create stacked area graph for Open Access data
+#' 
+#' @param df a data frame at the format produced by abm_oa_data()
+#' @return a ggplot object
+#' @import ggplot2 dplyr reshape2
+#' @export
+abm_graph_oadata_stackedarea <- function(df){
+  unpaywall_cols <- c("#F9BC01", "#8D4EB4", "#20E168", "#CD7F32", "#BBBBBB")
+  df_oa_graphdata <- df %>%
+    filter(Publication_Year_ch != "Total") %>%
+    select(Publication_Year_ch, gold_count, hybrid_count, green_count, bronze_count, closed_count) %>%
+    rename("Gold" = gold_count, "Hybrid" = hybrid_count, "Green" = green_count, "Bronze" = bronze_count, "Not OA" = closed_count)
+  
+  xymelt <- melt(df_oa_graphdata, id.vars = "Publication_Year_ch") %>%
```

Comment:
Could possibly use https://tidyr.tidyverse.org/articles/pivot.html functionality from tidyr instead of taking a dependency on reshape2, see for example https://tidyr.tidyverse.org/reference/pivot_longer.html